show_headnode test suite failure

```
___________________________________________________________________________ TestQuery.test_show_headnode ___________________________________________________________________________
self = <api.TestQuery instance at 0x7f727cf78560>, db = <sqlalchemy.orm.session.Session object at 0x7f727cf51c10>

    def test_show_headnode(self, db):
        api.project_create('anvil-nextgen')
        network_create_simple('spiderwebs', 'anvil-nextgen')
        api.headnode_create('BGH', 'anvil-nextgen', 'base-headnode')
        api.headnode_create_hnic('BGH', 'eth0')
        api.headnode_create_hnic('BGH', 'wlan0')
        api.headnode_connect_network('BGH', 'eth0', 'spiderwebs')
    
    
        result = json.loads(api.show_headnode('BGH'))
        # For the lists to be equal, the ordering must be the same:
        result['hnics'].sort()
>       assert result == {
            'name': 'BGH',
            'project': 'anvil-nextgen',
            'hnics': [
                'eth0',
                'wlan0',
            ],
            'vncport': None
        }
E       assert {'base_imgs':...nextgen', ...} == {'hnics': ['et...ncport': None}
E         Omitting 4 identical items, use -v to show
E         Left contains more items:
E         {u'base_imgs': u'base-headnode',
E          u'uuid': u'bce1ed76-b948-11e5-b2b0-001e65327848'}

tests/unit/api.py:1335: AssertionError
```

This looks like it was due to a recent change. About that CI...

Also, what was the reasoning behind putting the uuid in the output? It was originally just there as an implementation detail, to make sure that the libvirt names were unique; exposing it as part of the API doesn't make a lot of sense to me.

--- a/haas/api.py
+++ b/haas/api.py
@@ -837,7 +837,7 @@ def show_headnode(nodename):
         'hnics': [n.label for n in headnode.hnics],
         'vncport': headnode.get_vncport(),
         'uuid' : headnode.uuid,
-        'base_imgs': headnode.base_img,
+        'base_img': headnode.base_img,
     }, sort_keys = True)
 
 

--- a/tests/unit/api.py
+++ b/tests/unit/api.py
@@ -1346,6 +1346,7 @@ class TestQuery:
         api.project_create('anvil-nextgen')
         assert json.loads(api.list_project_nodes('anvil-nextgen')) == []
 
+    import uuid
 
     def test_show_headnode(self, db):
         api.project_create('anvil-nextgen')
@@ -1357,11 +1358,18 @@ class TestQuery:
 
 
         result = json.loads(api.show_headnode('BGH'))
+
+        # Verify UUID is well formed, then delete it, since we can't match it
+        # exactly
+        temp = uuid.UUID(result['uuid']) # Verify UUID is well-formed
+        del result['uuid']
+
         # For the lists to be equal, the ordering must be the same:
         result['hnics'].sort()
         assert result == {
-            'name': 'BGH',
-            'project': 'anvil-nextgen',
+            'name':     'BGH',
+            'project':  'anvil-nextgen',
+            'base_img': 'base-headnode',
             'hnics': [
                 'eth0',
                 'wlan0',

--- a/tests/unit/api.py
+++ b/tests/unit/api.py
@@ -1360,8 +1360,8 @@ class TestQuery:
         result = json.loads(api.show_headnode('BGH'))
 
         # Verify UUID is well formed, then delete it, since we can't match it
-        # exactly
-        temp = uuid.UUID(result['uuid']) # Verify UUID is well-formed
+        # exactly in the check below
+        temp = uuid.UUID(result['uuid'])
         del result['uuid']
 
         # For the lists to be equal, the ordering must be the same:

